connection.go

```diff
@@ -400,6 +400,8 @@ func (sc *snowflakeConn) queryContextInternal(
 		rows.addDownloader(populateChunkDownloader(ctx, sc, data.Data))
 	}
 
+	// SIG-16907: we occasionally panic on a nil value here, adding tracing to help diagnose.
+	logger.WithContext(ctx).Infof("Debug: rows chunk-downloader nil?: %v, is-multi-stmt? %v, err: %v", rows.ChunkDownloader == nil, isMultiStmt(&data.Data), err)
```

Comment:
maybe?

```
if rows == nil || rows.ChunkDownloader == null {
/// 
}}
```

Reply:
Sure, and in fact will break it down further.